test/chaat/handler/handler_test.clj

```diff
@@ -0,0 +1,77 @@
+(ns chaat.handler.handler-test
+  (:require [clojure.test :refer :all]
+            [chaat.handler.handler :as handler]
+            [java-time.api :as jt]
+            [chaat.model.user :as model.user]
+            [chaat.fixture :as fixture]))
+
+(use-fixtures :each fixture/test-fixture)
+
+(deftest health-check-test
+  (testing "Request to health check endpoint should return status 200"
+    (let [datasource (:db fixture/test-system)
+          request {}
+          response (handler/health-check datasource request)]
+      (is (= 200 (:status response))))))
+
+(deftest signup-test
+  (let [datasource (:db fixture/test-system)]
+    (testing "Missing username or password in request params, signup fails"
+      (let [params {:password "12345678"}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 400 (:status response)))))
+
+    (testing "Nil username or password in request params, signup fails"
+      (let [params {:username "john"
+                    :password nil}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 400 (:status response)))))
+
+    (testing "Username and password both present in request params, signup succeeds"
+      (let [params {:username "john"
+                    :password "12345678"}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 200 (:status response)))))
+
+      ;; perhaps no need to test this here? been tested at other layers already.
+      ;; depends on previous state, move to different block?
```

Comment:
[comment]
Where else has this been already tested? This does not look redundant to me. The other tests are testing either the model or DB layer, which are different interfaces (or black boxes, if you prefer that terminology). This applies to other similar comments you have made in this file.

It is quite possible that a bug is introduced in the request parsing down the line in the handler, which may not be caught by the other layers.

Reply:
Also remove this comment, once you have made a decision around this.

Reply:
> It is quite possible that a bug is introduced in the request parsing down the line in the handler, which may not be caught by the other layers.

Good point. Decided to test because of this, but was unsure because it was starting to feet a bit repetitive & redundant. This is clear now. Removed the comment.